### ISSUE TITLE
Got this error during training: terminate called after throwing an instance of 'c10::Error'

### ISSUE THREAD

**Issue by asrlhhh** [Totoro97/f2-nerf#19]
I encountered an issue while running the command 

```
python scripts/run.py --config-name=wanjinyou \
dataset_name=example case_name=ngp_fox mode=train \
+work_dir=$(pwd)
```

```
terminate called after throwing an instance of 'c10::Error'
  what():  Trying to create tensor with negative dimension -2036932816: [-2036932816, 10249059, 32766]
Exception raised from check_size_nonnegative at ../aten/src/ATen/EmptyTensor.h:15 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x6b (0x7f52f04e83cb in /home/ubuntu/f2-nerf/External/libtorch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0xce (0x7f52f04e3d9e in /home/ubuntu/f2-nerf/External/libtorch/lib/libc10.so)
frame #2: at::detail::empty_generic(c10::ArrayRef<long>, c10::Allocator*, c10::DispatchKeySet, c10::ScalarType, c10::optional<c10::MemoryFormat>) + 0x8ec (0x7f52c03c3dcc in /home/ubuntu/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #3: at::detail::empty_cpu(c10::ArrayRef<long>, c10::ScalarType, bool, c10::optional<c10::MemoryFormat>) + 0x55 (0x7f52c03c4e75 in /home/ubuntu/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #4: at::detail::empty_cpu(c10::ArrayRef<long>, c10::optional<c10::ScalarType>, c10::optional<c10::Layout>, c10::optional<c10::Device>, c10::optional<bool>, c10::optional<c10::MemoryFormat>) + 0x49 (0x7f52c03c4ef9 in /home/ubuntu/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #5: at::native::empty_cpu(c10::ArrayRef<long>, c10::optional<c10::ScalarType>, c10::optional<c10::Layout>, c10::optional<c10::Device>, c10::optional<bool>, c10::optional<c10::MemoryFormat>) + 0x33 (0x7f52c0ab1f63 in /home/ubuntu/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #6: <unknown function> + 0x249be31 (0x7f52c152ee31 in /home/ubuntu/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #7: at::_ops::empty_memory_format::redispatch(c10::DispatchKeySet, c10::ArrayRef<c10::SymInt>, c10::optional<c10::ScalarType>, c10::optional<c10::Layout>, c10::optional<c10::Device>, c10::optional<bool>, c10::optional<c10::MemoryFormat>) + 0x1b7 (0x7f52c11ec107 in /home/ubuntu/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #8: <unknown function> + 0x2473f98 (0x7f52c1506f98 in /home/ubuntu/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #9: at::_ops::empty_memory_format::call(c10::ArrayRef<c10::SymInt>, c10::optional<c10::ScalarType>, c10::optional<c10::Layout>, c10::optional<c10::Device>, c10::optional<bool>, c10::optional<c10::MemoryFormat>) + 0x15f (0x7f52c12274cf in /home/ubuntu/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #10: <unknown function> + 0xff531 (0x55f69fb79531 in /home/ubuntu/f2-nerf/build/main)
frame #11: <unknown function> + 0xfef81 (0x55f69fb78f81 in /home/ubuntu/f2-nerf/build/main)
frame #12: <unknown function> + 0x97b8f (0x55f69fb11b8f in /home/ubuntu/f2-nerf/build/main)
frame #13: <unknown function> + 0x1357f1 (0x55f69fbaf7f1 in /home/ubuntu/f2-nerf/build/main)
frame #14: <unknown function> + 0x51c01 (0x55f69facbc01 in /home/ubuntu/f2-nerf/build/main)
frame #15: __libc_start_main + 0xf3 (0x7f529182f083 in /lib/x86_64-linux-gnu/libc.so.6)
frame #16: <unknown function> + 0x5f5be (0x55f69fad95be in /home/ubuntu/f2-nerf/build/main)

Aborted (core dumped)
```

It looks like an incorrect CUDA setup or something...

My environment: 
Ubuntu 20.04
A100 GPU
native CUDA 11.6 (the one nvcc link to, does this matter? I don't quite understand if libtorch is using its own built-in cuda)
CUDNN 8.0
torch.__version__ '1.13.1+cu117'

**Comment by Diana-Zeng** [comment]
Hi,bro. I met the same situation as you.
My environment:
Ubuntu 18.04
NVIDIA 3090 GPU
native CUDA 11.1 
torch.version '1.10.1+cu111'
Have you already solved this problem?

**Comment by Diana-Zeng** [comment]
terminate called after throwing an instance of 'c10::IndexError'
  what():  Dimension out of range (expected to be in range of [-2, 1], but got 140698235630232)
Exception raised from maybe_wrap_dim at ../c10/core/WrapDimMinimal.h:33 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x6b (0x7ff7e049b0db in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libc10.so)
frame #1: <unknown function> + 0x14c2ed2 (0x7ff7302bded2 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #2: at::native::linalg_vector_norm(at::Tensor const&, c10::Scalar const&, c10::optional<c10::ArrayRef<long> >, bool, c10::optional<c10::ScalarType>) + 0xd4 (0x7ff7302c7514 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #3: <unknown function> + 0x2af3183 (0x7ff6e0c07183 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cuda_cu.so)
frame #4: <unknown function> + 0x2af3207 (0x7ff6e0c07207 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cuda_cu.so)
frame #5: at::_ops::linalg_vector_norm::redispatch(c10::DispatchKeySet, at::Tensor const&, c10::Scalar const&, c10::optional<c10::ArrayRef<long> >, bool, c10::optional<c10::ScalarType>) + 0xf3 (0x7ff730a66b73 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #6: <unknown function> + 0x3669056 (0x7ff732464056 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #7: <unknown function> + 0x366970c (0x7ff73246470c in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #8: at::_ops::linalg_vector_norm::call(at::Tensor const&, c10::Scalar const&, c10::optional<c10::ArrayRef<long> >, bool, c10::optional<c10::ScalarType>) + 0x1a5 (0x7ff730adba05 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #9: <unknown function> + 0x14ce786 (0x7ff7302c9786 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #10: at::native::linalg_norm_out(at::Tensor const&, c10::optional<c10::Scalar> const&, c10::optional<c10::ArrayRef<long> >, bool, c10::optional<c10::ScalarType>, at::Tensor&) + 0x41 (0x7ff7302ca9f1 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #11: at::native::linalg_norm(at::Tensor const&, c10::optional<c10::Scalar> const&, c10::optional<c10::ArrayRef<long> >, bool, c10::optional<c10::ScalarType>) + 0x109 (0x7ff7302cabc9 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #12: <unknown function> + 0x1fce737 (0x7ff730dc9737 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #13: at::_ops::linalg_norm::call(at::Tensor const&, c10::optional<c10::Scalar> const&, c10::optional<c10::ArrayRef<long> >, bool, c10::optional<c10::ScalarType>) + 0x1a5 (0x7ff7307eb5a5 in /home/diana/NeRF/f2-nerf/External/libtorch/lib/libtorch_cpu.so)
frame #14: <unknown function> + 0x7c838 (0x55bee9b77838 in /home/diana/NeRF/f2-nerf/build/main)
frame #15: <unknown function> + 0x7e2c6 (0x55bee9b792c6 in /home/diana/NeRF/f2-nerf/build/main)
frame #16: <unknown function> + 0x11029c (0x55bee9c0b29c in /home/diana/NeRF/f2-nerf/build/main)
frame #17: <unknown function> + 0x37ece (0x55bee9b32ece in /home/diana/NeRF/f2-nerf/build/main)
frame #18: __libc_start_main + 0xe7 (0x7ff6dbb96c87 in /lib/x86_64-linux-gnu/libc.so.6)
frame #19: <unknown function> + 0x455ca (0x55bee9b405ca in /home/diana/NeRF/f2-nerf/build/main)

Aborted (core dumped)

**Comment by yamal-shang** [comment]
I change -1 in every linalg_norm to std::vector<int64_t> dim = {-1}, and it works for me. -1 literal is not propertly converted into required IntArrayRef arg in linalg_norm, read https://github.com/pytorch/pytorch/issues/61027 for reference.

**Comment by Holmes-Alan** [comment]
> I change -1 in every linalg_norm to std::vector<int64_t> dim = {-1}, and it works for me. -1 literal is not propertly converted into required IntArrayRef arg in linalg_norm, read [pytorch/pytorch#61027](https://github.com/pytorch/pytorch/issues/61027) for reference.

How do you change them? Which files should I modify?

**Comment by Holmes-Alan** [comment]
> I change -1 in every linalg_norm to std::vector<int64_t> dim = {-1}, and it works for me. -1 literal is not propertly converted into required IntArrayRef arg in linalg_norm, read [pytorch/pytorch#61027](https://github.com/pytorch/pytorch/issues/61027) for reference.

it works, thank you!

**Comment by yuedajiong** [comment]
./src/Dataset/Dataset.cpp
./src/PtsSampler/PersSampler.cpp
./src/PtsSampler/PersSampler.cu

**Comment by Diana-Zeng** [comment]
> 我将每个 linalg_norm 中的 -1 更改为 std::vector<int64_t> dim = {-1}，它对我有用。-1 文字未正确转换为 linalg_norm 中所需的 IntArrayRef arg，请阅读[pytorch/pytorch#61027](https://github.com/pytorch/pytorch/issues/61027)以供参考。



> it works, thank you!
